terraform/main.tf

```diff
@@ -1,25 +1,139 @@
+      Component   = "backend"
+      ManagedBy   = "Terraform"
+      DeployedBy  = "Pipeline"
+    }
+  )
 }
 
 # Resource Group for the Job Application Backend
 resource "azurerm_resource_group" "backend" {
-  name     = var.resource_group_name
+  name     = local.resource_group_name
   location = var.location
 
-  tags = merge(
-    var.common_tags,
-    {
-      Purpose = "Backend-Infrastructure"
+  tags = local.common_tags
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+# Azure Container Registry for storing Docker images
+resource "azurerm_container_registry" "acr" {
+  name                = "acr${replace(local.naming_prefix, "-", "")}${var.environment}"
+  resource_group_name = azurerm_resource_group.backend.name
+  location            = azurerm_resource_group.backend.location
+  sku                 = var.acr_sku
+  admin_enabled       = false # Security best practice - use managed identity
+
+  # Enable quarantine policy for Premium SKU only (for vulnerability scanning)
+  quarantine_policy_enabled = var.acr_sku == "Premium" && var.environment == "prod" ? true : false
+
+  # Enable retention policy for production
+  retention_policy {
+    enabled = var.environment == "prod" ? true : false
+    days    = 30
+  }
+
+  # Trust policy (content trust) for Premium SKU
+  trust_policy {
+    enabled = var.acr_sku == "Premium" && var.environment == "prod" ? true : false
+  }
+
+  tags = merge(local.common_tags, {
+    Service = "ContainerRegistry"
+  })
+}
+
+# App Service Plan for hosting the backend API
+resource "azurerm_service_plan" "backend" {
+  name                = "asp-${local.naming_prefix}-backend-${var.environment}"
+  resource_group_name = azurerm_resource_group.backend.name
+  location            = azurerm_resource_group.backend.location
+
+  os_type  = "Linux"
+  sku_name = var.app_service_sku
+
+  tags = merge(local.common_tags, {
+    Service = "AppServicePlan"
+  })
+}
+
+# App Service for the backend API
+resource "azurerm_linux_web_app" "backend" {
+  name                = "app-${local.naming_prefix}-backend-${var.environment}"
+  resource_group_name = azurerm_resource_group.backend.name
+  location            = azurerm_service_plan.backend.location
+  service_plan_id     = azurerm_service_plan.backend.id
+
+  site_config {
+    always_on = var.environment == "prod" ? true : false
+
+    application_stack {
+      docker_image     = "${azurerm_container_registry.acr.login_server}/${var.docker_image_name}"
+      docker_image_tag = var.docker_image_tag
     }
-  )
+  }
+
+  # Application settings
+  app_settings = merge(var.app_settings, {
+    "WEBSITES_ENABLE_APP_SERVICE_STORAGE" = "false"
+    "DOCKER_REGISTRY_SERVER_URL"          = "https://${azurerm_container_registry.acr.login_server}"
+    "DOCKER_ENABLE_CI"                    = "true"
+    "NODE_ENV"                            = var.environment == "prod" ? "production" : "development"
+    "PORT"                                = "3001"
+  })
```

Comment:
Remove all this - we aren't using any of this stuff, would refer back to and read carefully what we're building; https://github.com/reespozzi/ai-academy-platforms/blob/main/part-4/task-1-terraform-container-apps.md